### PR TITLE
Parser created for iosxe show chassis

### PR DESCRIPTION
**Pull request by qduk**

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_chassis.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        p_header_1 = re.compile(r"^H\/W\s+Current$")
+
+        # Chassis#   Role    Mac Address     Priority Version  State                 IP
+        p_header_2 = re.compile(r"^Chassis#\s+Role\s+Mac\s+Address\s+Priority\s+Version\s+State\s+IP$")
+
+        # -------------------------------------------------------------------------------------
+        p_delimiter = re.compile(r"^-------------------------------------------------------------------------------------$")
+
+        # *1       Active   58bf.eacc.0c00     2      V02     Ready                169.254.130.6
+        p_chassis_count = re.compile(r"^(?P<slot>\S+)\s+(?P<role>\S+)\s+(?P<mac>\S+)\s+(?P<priority>\d+)\s+(?P<hw_version>\S+)\s+(?P<state>\S+)\s+(?P<ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})$")
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxe/show_chassis.py:74]: Please break long lines up

File: src/genie/libs/parser/iosxe/show_chassis.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                match = p_delimiter.match(line)
+                continue
+            # *1       Active   58bf.eacc.0c00     2      V02     Ready                169.254.130.6
+            elif p_chassis_count.match(line):
+                match = p_chassis_count.match(line)
+                if not chassis_obj.get("chassis_index"):
+                    chassis_obj.update({"chassis_index" : {}})
+                group = match.groupdict()
+                slot = int(group["slot"].replace("*", " ").strip())
+                chassis_obj["chassis_index"].update({ int(slot) : {"role" : group["role"], "mac_address" : group["mac"], "priority" : int(group["priority"]),
```
Comment (ThomasJRyan) [inline src/genie/libs/parser/iosxe/show_chassis.py:113]: Please clean this up

File: src/genie/libs/parser/iosxe/show_chassis.py (hunk 1 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_chassis.py]: Updated

File: src/genie/libs/parser/iosxe/show_chassis.py (hunk 2 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_chassis.py]: Updated

**Review by ThomasJRyan — APPROVED** [review]

**Pull request merged**